### PR TITLE
Fix realtime inference for mmcv models

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Move the `per_gpu_batch_size ` from image processor initialization to `get_collate_fn()`. In this way, batch size no longer needs to bound with image processor, which allows dynamic batch size in inference. 
2. Add `MMCV_MODELS` constant. We could put all the mmcv models here.
3. Improve the precision context manager.
4. Consider batch size in determining whether to use the realtime inference.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/tests/unittests/test_object_detection.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def test_mmdet_object_detection_inference(checkpoint_name):
         pipeline="object_detection",
     )
 
-    pred = predictor.predict({"image": [mmdet_image_name] * 10}, realtime=False)  # test batch inference
+    pred = predictor.predict({"image": [mmdet_image_name] * 10})  # test batch inference
```
Comment (bryanyzhu) [inline multimodal/tests/unittests/test_object_detection.py:38]: Is the default for realtime still False or True?

File: multimodal/src/autogluon/multimodal/utils/environment.py [hunk 2]
```diff
@@ -7,6 +8,11 @@
 import torch
 from torch import nn
 
+try:
+    from mmcv.parallel import DataContainer
+except:
+    pass
```
Comment (bryanyzhu) [inline multimodal/src/autogluon/multimodal/utils/environment.py:14]: Should we do `pass` or set mmcv to `None`?

File: multimodal/tests/unittests/test_object_detection.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/tests/unittests/test_object_detection.py:38]: The default is None. We infer whether True or Flase.

File: multimodal/src/autogluon/multimodal/utils/environment.py (hunk 2 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/utils/environment.py:14]: We don't use `mmcv` in this file. I think it doesn't matter setting None or not.

**Review by bryanyzhu — APPROVED** [review]
LGTM, thanks for the quick fix.

**Review by sxjscience — APPROVED** [review]
LGTM

**Pull request merged**